Formula/p/pyrefly.rb

```diff
@@ -0,0 +1,42 @@
+class Pyrefly < Formula
+  desc "Fast type checker and IDE for Python"
+  homepage "https://pyrefly.org/"
+  url "https://github.com/facebook/pyrefly/archive/refs/tags/0.18.0.tar.gz"
+  sha256 "fcc7529c577d6d8e9fcc53387b2af5c7252e41baba434a591b0c6dc4a35cc51c"
+  license "MIT"
+  head "https://github.com/facebook/pyrefly.git", branch: "main"
+
+  depends_on "rustup" => :build
+
+  def install
+    # rustup is keg-only in Homebrew, so we need to add it to PATH
+    ENV.prepend_path "PATH", Formula["rustup"].opt_bin
+    
+    # Install the required nightly toolchain
+    system "rustup", "toolchain", "install", "nightly-2025-03-29"
```

Comment:
this would be a bumper to include the formula to core. We dont like nightly build software.

Reply:
Is it better to use the `RUSTC_BOOTSTRAP=1` technique with the stable toolchain, or is that also disliked?

For reference: https://github.com/NixOS/nixpkgs/blob/de8144a7/pkgs/by-name/py/pyrefly/package.nix#L35

Reply:
> Is it better to use the `RUSTC_BOOTSTRAP=1` technique with the stable toolchain, or is that also disliked?

What does that option do? Our problem with rustup is: https://docs.brew.sh/Acceptable-Formulae#we-dont-like-install-scripts-that-download-unversioned-things

Reply:
As mentioned, `RUSTC_BOOTSTRAP` allows a stable toolchain to compile nightly features. The intended use case is for bootstrapping rustc but accomplishes what we want here. 

This won't be need long term, pyrefly only uses a couple unstable features and is working towards removing them: https://github.com/facebook/pyrefly/issues/374